lencode.go

```diff
@@ -78,7 +78,7 @@ func (e *Encoder) Encode(msg []byte) error {
 		return e.err
 	}
 
-	if len(msg) > math.MaxUint32 {
```

Comment:
I'd like to preserve the property that you can use the full 4 bytes for the message length. Instead of changing the left-hand side to MaxInt32 lets cast the right-hand side to an int64. That will allow us to support up to 2^32 byte payloads and should work on both 32 and 64 bit processors. Something like:
```
 	msgLen := int64(len(msg))
	if msgLen > math.MaxUint32 {
```

This will also require updating the `PutUint32` call:
```
	e.byteOrder.PutUint32(e.lenBuf[:], uint32(msg))
```